src/containers/SecretsModal/SecretsModal.js

```diff
@@ -72,14 +74,43 @@ export /* istanbul ignore next */ class SecretsModal extends Component {
     };
   }
 
-  componentDidUpdate(prevProps) {
+  componentDidUpdate(prevProps, prevState) {
     const { webSocketConnected } = this.props;
     const { webSocketConnected: prevWebSocketConnected } = prevProps;
     if (webSocketConnected && prevWebSocketConnected === false) {
       this.props.fetchServiceAccounts();
     }
+    if (this.state.name !== prevState.name) {
```

Comment:
It'd be useful to separate into the one function (e.g. shouldScroll that checks all of these things, passing in prevState and state), so it's tidier

Add a comment for said method please to explain why we need to do it

Reply:
done 😸